### DISCUSSION TITLE
Flashing error when trying to flash a older release 2021.06

### DISCUSSION THREAD

**Discussion by siddarth-murugesan** [bitcraze/discussions#466] [Q&A]
Hey Hi,

I am a master student from the University of Freiburg. and I am working on an obstacle avoidance project using the crazyflie. I am building and flashing using WSL2. I am able to build and flash via radio the current bitcraze firmware master with my local wsl environment, but I am only able to build the 2021.06 tag release and am not able to flash it via radio in the same environment. I am getting the error that cfloader is not found, as it is searching for it in Linux I guess. I haven't tried flashing it via the debugger yet. The commands used to build and try flashing the old tag are - make clean, make and make cload (as described in the demo examples in that tag branch)

(I am trying to test out the code of an external project - "A Modified EKF for Constant Altitude Flights" - https://www.bitcraze.io/2022/01/a-modified-ekf-for-constant-altitude-flights, which is over the 2021.06 release tag)

Thanks in advance

**Comment by krichardsson** [comment]
The cfloader is part of the [Crazyflie python client code](https://github.com/bitcraze/crazyflie-clients-python), you probably need to install it.

Looking at the code in the [pull request](https://github.com/bitcraze/crazyflie-firmware/pull/823), I think another option would be to try to merge it into the current codebase, those files have probably not been modified much since the pull request was created. I have not tried though and I might be wrong :-)

**Reply by siddarth-murugesan** [reply]
With my current environment, I am able to build and flash via radio, the current firmware master branch with commands - make cf2_defconfig, make -j 12 and make cload, but when I try to do the same as make clean, make and make cload over the older release tag 2021.06, then it shows up the error as cfloader is not found (even though I have it installed).

For now, I tried to merge it into the current codebase but facing some issues there as well.
Is the debugger the only way to flash the older release tag?

**Comment by knmcguire** [comment]
You could try to do it with the cfloader directly? Like this (for Ubuntu):

`python3 -m cfloader flash build/cf2.bin stm32-fw -w radio://0/20/2M/E7E7E7E701`

(might be that you need to do remove the 'build/' from 'build/cf2.bin)

Since the older release tag was before we converted to kbuild, I would advise cloning it in a separate repository, as switching after and before the kbuild era (which happened between 2022.1 and 2022.3) in the same repository is quite challenging.
